### ISSUE TITLE
Bug: SNM_MoveOrRemoveTrackFX  and FXID

### ISSUE THREAD

**Issue by hedacoder** [reaper-oss/sws#802]
```
selectedtrack = reaper.GetSelectedTrack(0,0)

FXID = reaper.TrackFX_GetFXGUID(selectedtrack, 0)
reaper.ShowConsoleMsg("\nFXID of FX 0 before moving = " .. FXID)

reaper.SNM_MoveOrRemoveTrackFX(selectedtrack, 0, 1) 

FXID = reaper.TrackFX_GetFXGUID(selectedtrack, 0)
reaper.ShowConsoleMsg("\nFXID of FX 0 after moving = " .. FXID)
```

REAPER 5.18, x64 win7, SWS 2.8.7
Run this lua code in a selected track with some FX in the chain
Symptom: Moving FX with the function SNM_MoveOrRemoveTrackFX doesn't move the FXID. After moving the effect down, the FXID of the same FX number, doesn't change. If you save and open the RPP file you can also see that everything changes correctly but FXID remains unmoved. 
see gif: http://i.imgur.com/21bDuI1.gif

**Comment by justinfrankel** [comment]
SNM_PreObjectState() calls RemoveAllIds() which removes the FXID lines, so the FXIDs aren't changed.

**Comment by justinfrankel** [comment]
I should also note that patching using "BYPASS" as a delimiter is a bit fragile... but I guess until we add an API to support native moving/deleting t's our fault anyway.

**Comment by nofishonfriday** [comment]
Thanks Justin.

I think it's a rather danegerous bug as it breaks scripts relying on FXID's to keep track of FX (eg [Stripper](https://forum.cockos.com/showthread.php?t=182233).

I made an attempt to fix (idea being swapping the FXID's after moving the FX), seems to work with the above provided test script,  but I'm not sure if it's a good way to do it.

So if you (or any of the other SWS collaborators) could take a look  I'd appreciate (mind you, I'm still a beginner... :))

https://github.com/nofishonfriday/sws/commit/ed3fa1f35db76b32001dfd44b455bfc0fef4fb7a

(I guess simply commenting the RemoveAllIds() call is not a good idea as it will probably break lots of other things ?)

**Comment by justinfrankel** [comment]
I will push a recommended fix for this issue on the next branch shortly.

**Comment by nofishonfriday** [comment]
I've added your fix to the whatsnew, if you don't mind.
Thanks again.

**Comment by nofishonfriday** [comment]
fixed in v2.9.8

**Comment by nofishonfriday** [comment]
fixed in v2.9.8
